content/blog/2019-11-13-llvm-function-inlining/index.md

```diff
@@ -0,0 +1,144 @@
++++
+title = "LLVM Function Inlining Pass"
+[extra]
+bio = """
+  Gautam Mekkat and Mark Anastos are undergraduate seniors studying Computer
+  Science and Electrical & Computer Engineering.
+"""
+[[extra.authors]]
+name = "Gautam Mekkat"
+link = "https://github.com/Blue9"
+[[extra.authors]]
+name = "Mark Anastos"
+link = "https://github.com/anastos"
++++
+
+It is a common convention in computer programming to abstract out repeated
+functionality in a program into procedures, so that it can be referenced
+multiple times without having to repeat the code. Separating out code in this
+way can increase the legibility and modularity of the program, but comes at a
+cost. Namely, function calls in programs can introduce a major runtime
+performance overhead due to the costs of entering and exiting functions.
+Typically, when a function is called, a new stack frame will need to be created
+and the instruction pointer will need to be moved to the beginning of that
+function—which may or may not be in the instruction cache—among other
+function-initialization tasks. Also, some of the state of the calling function
+(e.g., caller-saved registers) will need to be stored in preparation for the
+call, such that it will not be overwritten by the called function. Separating
+out functions also gives the compiler less information for optimizing the
+program, as, when looking at a function, the compiler cannot be sure of the
```

Comment:
I think this sentence would be a little easier to read if you replaced "as" with "because"?